hashintel/hash
# HASH Sidebar updates

## 🌟 What is the purpose of this PR?



This PR  addresses follow up tasks pointed out in #367 

## ⚠️ Known issues


- Bookmarks section not implemented
- Drag/Drop/Reordering functionality in pages section not implemented

## 🚫 Blocked by





## 🐾 Next steps




## 🔍 What does this change?




- Removes `CreatePageModal` is rendered. This is because in the new flow, we no longer show the modal but instead create a new page with title "Untitled" and navigate the user to the page.
- Update `WorkspaceSwitcher` to show correct details of current workspace
- Adds `material-ui-popup-state` to make it easier to connect menus/popovers to triggers and handle the menus in general
- Updates hover/active/focus styles of Sidebar nav links
- Updates focus styles of `IconButtons` including `HeaderIconButton`
- Adds default styling for MUI's `Menu` and `MenuItem` components and switches existing usages of Popover to Menu. This addresses => https://github.com/hashintel/hash/pull/367#discussion_r819721840
- Updates Button/IconButton to use `:focus-visible` for handling focus styling instead of `:focus`
- Updates how `sx` prop is handled in components
- Adds the UI for page menu and type menu
- Exports 2 new methods `createUntitledPage` and `createSubPage` from `useCreatePage`. 
- Adds client search/ ordering functionality for Types.
- Adds `CurrentWorkspaceContext` for getting the accountId of the workspace the user is currently on

### 📜 Does this require a change to the docs?



- No

## 🔗 Related links





- [Sidebar updates](https://app.asana.com/0/0/1201926880408104) _(internal)_
- [Update Pages and Types Submenu w/ functionality](https://app.asana.com/0/1201927504137297/1201926880408107) _(internal)_
- [Get Clarification on HASH button designs](https://app.asana.com/0/1201927504137297/1201970018157947) _(internal)_
- [Fix how we accept `sx` prop across the hash app](https://app.asana.com/0/1201959586244671/1201988634400677)

## 🛡 What tests cover this?



- Updated page creation playwright test
- Manual tests

## ❓ How to test this?



1.  Checkout the branch and install latest dependencies
1.  Login and confirm the following works
1.  - Switching workspaces
1.  - Create Page / Create Entity
1.  - Page submenu shows up / Create Subpage works / Copy page url link works
1.  - Entity submenu shows up / Create Type works / Copy Type url link works
1.  - Focus states show up properly

## 📹 Demo



https://user-images.githubusercontent.com/16928688/159611299-a495d0c5-aacf-4e08-8b01-ed4957ab0575.mp4



---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1201988634400677
  - https://app.asana.com/0/0/1201926880408104
  - https://app.asana.com/0/0/1201926880408107
  - https://app.asana.com/0/0/1201970018157947
